designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala

```diff
@@ -122,14 +185,161 @@ class UsageStatisticsReportsSettingsDeterminer(
+
+          //        Fragment stats
+          val fragments        = scenariosInputData.map(_.fragmentsUsed).sorted
+          val fragmentsMedian  = fragments.get(fragments.length / 2).getOrElse(0)
+          val fragmentsAverage = fragments.sum / scenarios
+
+          Map(
+            "m_n" -> nodesMedian,
+            "v_n" -> nodesAverage,
+            "c"   -> categories,
+            "v_m" -> versionsMedian,
+            "v_v" -> versionsAverage,
+            "a_n" -> authors,
+            "m_f" -> fragmentsMedian,
+            "v_f" -> fragmentsAverage,
+          ).mapValuesNow(_.toString)
+      }
+    val activityQueryParams = fetchActivity()
+      .map {
+        case List()           => Map.empty[String, String]
+        case listOfActivities =>
+          //        Attachment stats
+          val attachments = listOfActivities
+            .map { activity =>
+              activity.attachments
+            }
+            .map(_.length)
+            .sorted
+          val attachmentMedian  = attachments.get(attachments.length / 2).getOrElse(0)
+          val attachmentAverage = attachments.sum / attachments.length
+          val attachmentExist   = attachments.count(_ > 0)
+
+          //        Comment stats
+          val comments = listOfActivities
+            .map { activity =>
+              activity.comments
+            }
+            .map(_.length)
+            .sorted
+          val commentsMedian  = comments.get(comments.length / 2).getOrElse(0)
+          val commentsAverage = comments.sum / comments.length
+
+          Map(
+            "m_a" -> attachmentMedian,
+            "v_a" -> attachmentAverage,
+            "e_a" -> attachmentExist,
+            "m_c" -> commentsMedian,
+            "v_c" -> commentsAverage
+          ).mapValuesNow(_.toString)
+      }
+
+    val componentStats =
+      fetchComponentList().map {
+        case List() => Map.empty[String, String]
+        case componentList =>
+          val withoutFragments = componentList.filterNot(comp => comp.componentType == ComponentType.Fragment)
+          val byName           = withoutFragments.groupBy(_.name)
+
+          Map(
+            "c_n" -> byName.size.toString
+          )
+      }
+
+    val uptimeStats =
+      fetchScenarioActions().map {
+        case List() => Map.empty[String, String]
+        case listForScenarios =>
+          val uptimes = listForScenarios.map { listOfActions =>
+            val durationsOfDeployment = calculateUpTimeStats(listOfActions).filter(_.isDefined).sequence.get.sorted
+            //         return average running time for this scenario
+            if (durationsOfDeployment.nonEmpty) {
+              durationsOfDeployment.sum / durationsOfDeployment.length
+            } else {
+              0
+            }
+
+          }
+          val uptimeAverage = uptimes.sum / uptimes.length
+          val uptimeMedian  = uptimes.get(uptimes.length / 2).get
+          Map(
+            "m_u" -> uptimeMedian,
```

Comment:
IMHO median for uptime does not make sense. Instead of this, I would add min and max